packages/extension/src/contexts/LedgerContext/LedgerContext.tsx

```diff
@@ -729,44 +729,59 @@ const LedgerProvider: FC = ({ children }) => {
       signOnly = false
     ): Promise<SubmitTransactionResponse | SignTransactionResponse> => {
       const wallet = getCurrentWallet();
-      if (!client) {
-        throw new Error('You need to be connected to a ledger');
-      }
       if (!wallet) {
         throw new Error('You need to have a wallet connected');
       }
+      if (!signOnly) {
+        if (!client) {
+          throw new Error('You need to be connected to a ledger');
+        }
+      }
 
+      // Ensure the transaction has an Account or set it to the wallet's public address
       if (!payload.transaction.Account || payload.transaction.Account === '') {
         payload.transaction.Account = wallet.publicAddress;
       }
 
       // Validate the transaction
       validate(payload.transaction as unknown as Record<string, unknown>);
-      // Prepare the transaction
-      const prepared: Transaction = await client.autofill(payload.transaction);
+
+      // Prepare the transaction if needed
+      let prepared: Transaction = payload.transaction;
+      if (!signOnly && client) {
```

Comment:
if (!signOnly) {

Reply:
Same comment than above, if we have a change on line 735, this could create a bug.
If I replace this line, I would have to write:
`prepared = await client?.autofill(payload.transaction);`
But in that case prepared could be undefined.

Reply:
fine but it should not change